### DISCUSSION TITLE
RPi3 + Node-red + zwavejs + RaZberry 7 issues

### DISCUSSION THREAD

**Discussion by Sharppis1** [zwave-js/node-red-contrib-zwave-js#256] [Q&A]
I have setup with node-red running on rpi3+ and I have installed zwavejs (from Manage palette) to control my RaZberry 7. First thing, is this compatible with zwavejs? I'm not sure because in config db under z-wave.me you can find couple different RazBerries, one is 7 pro, which isn't exactly my version but 7 anyway. So I assume that it _should_ work.

Couple basic information of my setup to help debugging.

RazBerry can be found behind `/dev/ttyAMAO`

OS - version:
PRETTY_NAME="Raspbian GNU/Linux 11 (bullseye)"
NAME="Raspbian GNU/Linux"
VERSION_ID="11"
VERSION="11 (bullseye)"
VERSION_CODENAME=bullseye

RPI3+ /boot/config.txt does have `enable_uart=1
` and there's also setting `dtoverlay=disable-bt`

Raspbian is fresh installation (today) as well as node-red and all libraries etc. So it should be quite okay base setup. 

Node-red starts and loads zwavejs modules, so everything looks fine, but after init process, status of Controller node turns in to yellow and it says Nodes : 2,10,11 Not ready. 

Debug window says: 
`msg.payload : Object
object
networkId: 1
node: 2
event: "INTERVIEW_FAILED"
timestamp: 1662756144467
object: object
errorMessage: "The node is dead"
isFinal: true`

The node is dead does sound a bit scary. But I think it does some communication with RaZberry because from zwave-js.log I can see that it has received API capabilities list and all kinds of data from control (CNTRLR). Finally it ends with:

2022-09-09T20:44:37.769Z CNTRLR   setting serial API timeouts: ack = 1000 ms, byte = 150 ms
2022-09-09T20:44:37.811Z CNTRLR   serial API timeouts overwritten. The old values were: ack = 1000 ms, byte = 15
2022-09-09T20:44:37.812Z CNTRLR   Interview completed
2022-09-09T20:44:37.820Z CNTRLR   [Node 001] The node is alive.
2022-09-09T20:44:37.821Z CNTRLR   [Node 001] The node is ready to be used
2022-09-09T20:44:37.825Z CNTRLR   [Node 002] Beginning interview - last completed stage: ProtocolInfo
2022-09-09T20:44:37.825Z CNTRLR » [Node 002] pinging the node...
2022-09-09T20:44:37.835Z CNTRLR   [Node 010] The node is asleep.
2022-09-09T20:44:37.841Z CNTRLR   [Node 010] Beginning interview - last completed stage: ProtocolInfo
2022-09-09T20:44:37.842Z CNTRLR » [Node 010] querying node info...
2022-09-09T20:44:37.850Z CNTRLR   [Node 011] The node is asleep.
2022-09-09T20:44:37.855Z CNTRLR   [Node 011] Beginning interview - last completed stage: ProtocolInfo
2022-09-09T20:44:37.856Z CNTRLR » [Node 011] querying node info...
2022-09-09T20:44:42.131Z CNTRLR   [Node 002] The node did not respond after 1 attempts, it is presumed dead
2022-09-09T20:44:42.133Z CNTRLR   [Node 002] The node is dead.

I have tried so many different settings, configs and almost some black magic, but still no success. I have done something right because in the first place controller node didn't even turn into yellow and I had some weird permission problems, I managed pass those problems. **Any help with this would be great!**

**Comment by marcus-j-davies** [comment]
Hi @Sharppis1,

> First thing, is this compatible with zwavejs?

RaZberry 7 should be fine - ZWave transceivers that use the zwave serial API bus is supported.

> Node-red starts and loads zwavejs modules, so everything looks fine, but after init process, status of Controller node turns in to yellow and it says Nodes : 2,10,11 Not ready.

Any node that is not ready (a thumbs up in the UI) means it's not been interviewed fully, or if its mains powered, has not yet responded to a ping during Start up.

Like all zwave software, devices need to be interviewed, to understand what features/capabilities they have, Node RED is no different.
However, the devices needs to be awake for an interview to happen.

For the nodes that are not ready, try waking them up (if battery operated) - so they get interviewed.
Note: Triggering their sensor is not waking them up - most devices have a wake up routine by some button press of sorts.

Waking them up should cause an interview, after one succeeds wake up the other (this is only a I time thing)
Its part of the inclusion process - but since nodes have already been added outside of Node RED - its playing catchup

> msg.payload : Object object networkId: 1 node: 2 event: "INTERVIEW_FAILED" timestamp: 1662756144467 object: object errorMessage: "The node is dead" isFinal: true

This one is a little more serious.
This suggests the device (node 2) is not responding to a basic ping request (NOP command).
was this previously added in secure mode before you brought it over to Node RED?

If so - have you also provided the same Keys in the Controller config that was used to include them?
